### PR TITLE
Record handler interface

### PR DESCRIPTION
**Pull request by tuhrig**

### CODE REVIEW

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisRecordProcessor.kt [hunk 1]
```diff
@@ -105,7 +104,7 @@ class AwsKinesisRecordProcessor<D, M, C : Class<KinesisEvent<D, M>>>(private val
     }
 
     override fun shutdown(shutdownInput: ShutdownInput?) {
-        log.info("Shutting down record processor")
+        log.info("Shutting down record build")
```
Comment (jmatusewicz-bringmeister) [inline src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisRecordProcessor.kt]: I think intellij refactored here bit too much

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/KinesisListener.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            return DefaultKinesisListener(streamName, dataClass, metadataClass, eventProcessor)
+        }
+
+        inline fun <reified D, reified M> build(streamName: String, noinline eventProcessor: (D, M) -> Unit): KinesisListener<D, M> {
+            return DefaultKinesisListener(streamName, D::class.java, M::class.java, eventProcessor)
+        }
+    }
+
+    fun streamName() : String
+
```
Comment (jmatusewicz-bringmeister) [inline src/main/kotlin/de/bringmeister/spring/aws/kinesis/KinesisListener.kt]: lets get rid of the empty lines between the interface methods :-)

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/WorkerFactory.kt [hunk 3]
```diff
@@ -6,14 +6,13 @@ import com.amazonaws.services.kinesis.clientlibrary.lib.worker.Worker
 import com.fasterxml.jackson.databind.ObjectMapper
 import java.util.concurrent.TimeUnit
 
-class WorkerFactory(private val objectMapper: ObjectMapper) {
+class WorkerFactory(val objectMapper: ObjectMapper) {
```
Comment (jmatusewicz-bringmeister) [inline src/main/kotlin/de/bringmeister/spring/aws/kinesis/WorkerFactory.kt]: can the objectMapper be private again?

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisRecordProcessor.kt (hunk 1 above)
Comment (tuhrig) [reply src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisRecordProcessor.kt]: Ups...

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/KinesisListener.kt (hunk 2 above)
Comment (tuhrig) [reply src/main/kotlin/de/bringmeister/spring/aws/kinesis/KinesisListener.kt]: ok

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/WorkerFactory.kt (hunk 3 above)
Comment (tuhrig) [reply src/main/kotlin/de/bringmeister/spring/aws/kinesis/WorkerFactory.kt]: sure

**Pull request merged**